src/MICore/SetMIDebugLogging.cmd

```diff
@@ -42,68 +42,83 @@ if NOT "%~1"=="" goto :ArgLoop
 if "%LoggingValue%"=="" echo ERROR: 'on' or 'off' must be specified.& exit /b -1
 if /i NOT "%LoggingValue%"=="1" if NOT "%ServerLogging%"=="" echo ERROR: '/serverlogging' can only be used with 'on'& exit /b -1
 
-set reg_exe=reg.exe
-if /i "%PROCESSOR_ARCHITECTURE%"=="AMD64" set reg_exe=%windir%\syswow64\reg.exe
-
-set MainRegRoot=HKLM\SOFTWARE\Microsoft\VisualStudio\%VSVersion%
-set OptionalRegRoots=HKCU\SOFTWARE\Microsoft\VisualStudio\%VSVersion%_Config HKLM\SOFTWARE\Microsoft\Glass\%VSVersion%
-
-if "%Exp%"=="1" goto :SetExpRegRoots
-
-call :SetRegRoot %MainRegRoot%
-for %%r in (%OptionalRegRoots%) do call :SetForRegRootIfExists %%r
-exit /b 0
-
-:SetExpRegRoots
-reg query HKCU\Software\Microsoft\VisualStudio | findstr %VSVersion%Exp_Config > %TMP%\%VSVersion%ExpRegRoots.txt
-set ExpRoots=
-set ExpRootFound=
-for /f "delims=" %%r in (%TMP%\%VSVersion%ExpRegRoots.txt) do set ExpRootFound=1& call set ExpRoots=%%ExpRoots%% %%r
-if NOT "%ExpRootFound%"=="1" echo ERROR: Experimental instance could not be found. Experimental instance must be started before this batch script can work.& exit /b -1
-
-for %%r in (%ExpRoots%) do call :SetRegRoot %%r
-exit /b 0
-
-:SetForRegRootIfExists
-REM check if the registry root exists
-call %reg_exe% query %1 /ve 1>NUL 2>NUL
-if NOT %ERRORLEVEL%==0 goto eof
-
-:SetRegRoot
-call %reg_exe% add %1\Debugger /v EnableMIDebugLogger /t REG_DWORD /d %LoggingValue% /f
-if not %ERRORLEVEL%==0 echo ERROR: failed to write to the registry (%1) & exit /b -1
-
-if "%ServerLogging%"=="" goto DeleteServerLogging
-goto AddServerLogging
-
-:AddServerLogging
-call %reg_exe% add %1\Debugger /v GDBServerLoggingArguments /t REG_SZ /d "%ServerLogging%" /f
-if not %ERRORLEVEL%==0 echo ERROR: failed to write to the registry (%1) & exit /b -1
-goto eof
-
-:DeleteServerLogging
-call %reg_exe% delete %1\Debugger /v GDBServerLoggingArguments /f 1>NUL 2>NUL
-goto eof
+set SetLoggingError=
+if NOT "%VSRootDir%"=="" call :SetLogging "%VSRootDir%" & goto Done
+
+set ProgRoot=%ProgramFiles(x86)%
+if "%ProgRoot%"=="" set ProgRoot=%ProgramFiles%
+
+set VSVersionFound=
+set MIEngineFound=
+call :TryVSPath "%ProgRoot%\Microsoft Visual Studio 14.0"
+call :TryVSPath "%ProgRoot%\Microsoft Visual Studio\VS15Preview"
```

Comment:
Do we want to enable/disable logging for Dev14 & Dev15 at the same time. (I'm fine with that).